contracts/utils/Zapper.sol

```diff
@@ -0,0 +1,230 @@
+            INonfungiblePositionManager.CollectParams({
+                tokenId: tokenId,
+                recipient: address(this),
+                amount0Max: 2**128 - 1,
+                amount1Max: 2**128 - 1
+            })
+        );
+
+        nonfungiblePositionManager.burn(tokenId);
+
+        if (tokenOut != token0) {
+            ERC20Helper._approveToken(token0, address(swapRouter), amount0);
+
+            amount0 = swapRouter.exactInputSingle(
+                ISwapRouter.ExactInputSingleParams({
+                    tokenIn: token0,
+                    tokenOut: tokenOut,
+                    fee: _findBestFee(token0, tokenOut),
+                    recipient: address(this),
+                    deadline: block.timestamp + 1,
+                    amountIn: amount0,
+                    amountOutMinimum: 1,
+                    sqrtPriceLimitX96: 0
+                })
+            );
+        }
+
+        if (tokenOut != token1) {
+            ERC20Helper._approveToken(token1, address(swapRouter), amount1);
+
+            amount1 = swapRouter.exactInputSingle(
+                ISwapRouter.ExactInputSingleParams({
+                    tokenIn: token1,
+                    tokenOut: tokenOut,
+                    fee: _findBestFee(tokenOut, token1),
+                    recipient: address(this),
+                    deadline: block.timestamp + 1,
+                    amountIn: amount1,
+                    amountOutMinimum: 1,
+                    sqrtPriceLimitX96: 0
+                })
+            );
+        }
+
+        ERC20Helper._approveToken(tokenOut, address(this), amount0 + amount1);
+        ERC20Helper._withdrawTokens(tokenOut, msg.sender, amount0 + amount1);
+    }
+
+    ///@notice orders token addresses
+    ///@param token0 address of first token
+    ///@param token1 address of second token
+    ///@return address ordered token address
+    ///@return address ordered token address
+    function _reorderTokens(address token0, address token1) internal pure returns (address, address) {
+        if (token0 > token1) {
+            return (token1, token0);
+        } else {
+            return (token0, token1);
+        }
+    }
+
+    ///@notice finds the best fee tier on which to perform a swap
+    ///@param token0 address of first token
+    ///@param token1 address of second token
+    ///@return fee suggested fee tier
+    function _findBestFee(address token0, address token1) internal view returns (uint24 fee) {
+        uint128 bestLiquidity = 0;
+        uint16[4] memory fees = [100, 500, 3000, 10000];
+
+        for (uint8 i = 0; i < 4; i++) {
+            try this.getPoolLiquidity(token0, token1, uint24(fees[i])) returns (uint128 nextLiquidity) {
+                if (nextLiquidity > bestLiquidity) {
+                    bestLiquidity = nextLiquidity;
+                    fee = fees[i];
+                }
+            } catch {
+                //pass
```

Comment:
What happens if we get into this catch block?

Reply:
Nothing! If a pool with those tokens and fee doesn't exist, we can't use that and we check for the next one. If no pool exists, than bestLiquidity is never updated and we end in the if block => the transaction reverts